icu4c/source/test/intltest/tzfmttst.cpp

```diff
@@ -1308,6 +1309,30 @@ TimeZoneFormatTest::TestFormatCustomZone() {
     }
 }
 
+void
+TimeZoneFormatTest::Test22615NonASCIIID() {
+    UErrorCode status = U_ZERO_ERROR;
+    LocalPointer<TimeZoneNames> tzdb(TimeZoneNames::createTZDBInstance(Locale("en"), status));
+    // A test to ensure under the debugging build non ASCII id will not cause
+    // internal assertion error.
+    UnicodeString id(9, u'\u00C0', 8);
+    UnicodeString output;
+    tzdb->getMetaZoneDisplayName(id, UTZNM_SHORT_STANDARD, output);
+    status = U_ZERO_ERROR;
+    std::unique_ptr<icu::StringEnumeration> enumeration(
+        tzdb->getAvailableMetaZoneIDs(id, status));
+    output.remove();
+    tzdb->getMetaZoneID(id, 0, output);
+    output.remove();
+    tzdb->getMetaZoneDisplayName(id, UTZNM_EXEMPLAR_LOCATION, output);
+    output.remove();
+    tzdb->getTimeZoneDisplayName(id, UTZNM_SHORT_DAYLIGHT, output);
+    output.remove();
+    tzdb->getExemplarLocationName(id, output);
+    output.remove();
+    tzdb->getDisplayName(id, UTZNM_LONG_GENERIC, 0, output);
```

Comment:
I'm a little uncomfortable with a unit test that calls a bunch of stuff and doesn't check the results.  While "it doesn't crash" is a reasonable success criterion, it seems like you should be taking some kind of stand as to what the code _should_ do when passed an invalid value like this.

In particular, you're calling a bunch of functions that don't return an error code, so you can't just check the error code.  If they can't signal an error and you pass then an invalid ID, what should they do?  I can't tell that from this unit test.

Reply:
I check the value now